### PR TITLE
limit indexing count and metrics for discarded index request.

### PR DESCRIPTION
**Pull request by manan164**
This PR contains changes to limit the number of indexing call we make.
Also I have added metrics for number of discarded request.

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java [hunk 1]
```diff
@@ -169,7 +169,10 @@ public long getPendingWorkflowCount(String workflowName) {
     public String createWorkflow(Workflow workflow) {
         workflow.setCreateTime(System.currentTimeMillis());
         executionDAO.createWorkflow(workflow);
-        indexDAO.asyncIndexWorkflow(workflow);
+        if (workflow.getStatus().isTerminal()) {
```
Comment (naveenchlsn) [inline core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java]: Is it possible we have nullpointer exception here? Can we add some checks if it is possible ?

**Comment by naveenchlsn** [comment]
> This PR contains changes to limit the number of indexing call we make.
> Also I have added metrics for number of discarded request.

How many calls are made now vs previous ? 
Can the workflow be viewed in dashboard with these indexes ?

File: core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java [hunk 2]
```diff
@@ -270,7 +276,10 @@ public void updateTask(Task task) {
                 }
             }
             executionDAO.updateTask(task);
-            indexDAO.asyncIndexTask(task);
+            if (task.getStatus().isTerminal()) {
```
Comment (naveenchlsn) [inline core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java]: same as above. Please check for NLP as that might break the flow.

File: core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java (hunk 1 above)
Comment (manan164) [reply core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java]: No. Any status will have compulsory three values.https://github.com/Swiggy/conductor/blob/master/common/src/main/java/com/netflix/conductor/common/metadata/tasks/Task.java#L53

**Comment by manan164** [comment]
> > This PR contains changes to limit the number of indexing call we make.
> > Also I have added metrics for number of discarded request.
> 
> How many calls are made now vs previous ?
> Can the workflow be viewed in dashboard with these indexes ?

I am expecting the indexing calls to be half of what previously it used to come as we are not indexing when status is not terminal. For a workflow having single task we used to make 6 calls each execution. Now we will make only 3 calls.

File: core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java (hunk 1 above)
Comment (naveenchlsn) [reply core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java]: What about status itself. Can workflow have status field set as null ?

**Comment by manan164** [comment]
Closing this PR as it is not required.

**Pull request closed without merging**